### PR TITLE
feat: LSDV-4583-4: Image per item segmentation

### PR DESCRIPTION
**Pull request by nicholasrq**
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [x] Frontend

### Describe the reason for change
LSDV-4583-4

#### What is the new behavior?
With this PR users can perform image segmentation on per-image basis.

#### What is the current behavior?
Previously it was only possible to segment layers, so segmentation is shared across all images in a set.

#### What libraries were added/updated?
None

#### Does this change affect performance?
No

#### Does this change affect security?
No

#### What alternative approaches were there?
None

#### What feature flags were used to cover this change?
`feat_front_lsdv_4583_multi_image_segmentation_short`

### Does this PR introduce a breaking change?
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
- [x] e2e
- [ ] integration
- [ ] unit

### Which logical domain(s) does this change affect?
Labeling Interface, Image Segmentation

### CODE REVIEW

File: src/tags/object/Image/ImageSelection.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+      return self.isActive ? {
+        left: Math.min(start.x, end.x),
+        top: Math.min(start.y, end.y),
+        right: Math.max(start.x, end.x),
+        bottom: Math.max(start.y, end.y),
+      } : null;
+    },
+    includesBbox(bbox) {
+      return self.isActive && bbox && self.bbox.left <= bbox.left && self.bbox.top <= bbox.top && self.bbox.right >= bbox.right && self.bbox.bottom >= bbox.bottom;
```
Comment (yyassi-heartex) [inline src/tags/object/Image/ImageSelection.js:47]: minor tweak - to make it a little more readable
```suggestion
      if (!self.isActive || !bbox) return false;
      const isLeftOf = self.bbox.left <= bbox.left;
      const isAbove = self.bbox.top <= bbox.top;
      const isRightOf = self.bbox.right >= bbox.right;
      const isBelow = self.bbox.bottom >= bbox.bottom;
      
      return isLeftOf && isAbove && isRightOf && isBelow;
```

**Review by bmartel — APPROVED** [review]

File: src/stores/Annotation/Annotation.js [hunk 2]
```diff
@@ -1150,6 +1150,8 @@ export const Annotation = types
             value,
           };
 
+          console.log('create new area', areaSnapshot);
+
```
Comment (yyassi-heartex) [inline src/stores/Annotation/Annotation.js:1154]: please remove the console.log
```suggestion
```

**Review by yyassi-heartex — APPROVED** [review]
looking good - just a couple of minor things for readability - otherwise approved

File: src/tags/object/Image/ImageSelection.js (hunk 1 above)
Comment (nicholasrq) [reply src/tags/object/Image/ImageSelection.js:47]: nice! i'm not the author of this code, just split a giant `Image.js` into smaller pieces. gonna apply this one anyways. thanks

**Review by hlomzik — APPROVED** [review]
remove `console.log`s. otherwise it's cool!

File: src/stores/RegionStore.js [hunk 3]
```diff
@@ -121,7 +124,6 @@ const SelectionMap = types.model(
     highlight(region) {
       self.clear();
       self.select(region);
-      region?.shapeRef?.parent?.canvas?._canvas?.scrollIntoView?.();
```
Comment (hlomzik) [inline src/stores/RegionStore.js:124]: why do you remove it? it was helpful for multiple images in config to scroll to the respective image when user selects some region.

**Pull request closed without merging**